spec/services/distribution_create_service_spec.rb

```diff
@@ -0,0 +1,57 @@
+RSpec.describe DistributionCreateService, type: :service do
+  subject { DistributionCreateService }
+  describe "call" do
+    let!(:storage_location) { create(:storage_location, :with_items) }
+    let!(:distribution_params) { { organization_id: @organization.id, partner_id: @partner.id, storage_location_id: storage_location.id, line_items_attributes: { "0": { item_id: storage_location.items.first.id, quantity: 5 } } } }
+
+    it "replaces a big distribution with a smaller one, resulting in increased stored quantities" do
+      expect do
+        subject.new(distribution_params).call
+      end.to change { storage_location.reload.size }.by(-5)
+    end
+
+    it "returns a successful object with Scheduled distribution" do
+      result = subject.new(distribution_params).call
+      expect(result).to be_instance_of(subject)
+      expect(result).to be_success
+      expect(result.distribution).to be_scheduled
+    end
+
+    it "Sends a PartnerMailer" do
+      expect(PartnerMailerJob).to receive(:perform_async).once
+      allow(Flipper).to receive(:enabled?).with(:email_active).and_return(true)
+      subject.new(distribution_params).call
+    end
+
+    context "when provided with a request ID" do
+      let!(:request) { create(:request) }
```

Comment:
WARNING: Severely opinionated nitpick incoming

Since this object is only used withing this one `context` block, could it just be set as a variable in the `it` block? I tend to feel like `let` magic makes specs hard to read when they aren't needed. My reasons why are best articulated in [this](https://thoughtbot.com/blog/lets-not) blog post.

I get that using them is part of the house style for this project, so I dont want to make a big push to rewrite things, but in this case the object is only used in this one spec, so the `let` feels a little unnecessary.

Reply:
I was expecting you to chime in about this ;) 

Let's (no pun intended) hold off, because there's some additional refactoring that needs to happen and I think a lot of these can be extracted to a shared example; the abstraction will hopefully facilitate that.

Please call me out on it if this persists unnecessarily after the additional refactoring